src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -1,65 +1,525 @@
+      m_modules[0].getDriveVelocity(),
+      m_modules[1].getDriveVelocity(),
+      m_modules[2].getDriveVelocity(),
+      m_modules[3].getDriveVelocity()
+    };
+  }
+  public double[] getSteerVelocities(){
+    return new double[] {
+      m_modules[0].getSteerVelocity(),
+      m_modules[1].getSteerVelocity(),
+      m_modules[2].getSteerVelocity(),
+      m_modules[3].getSteerVelocity()
+    };
+  }
+  public void runCharacterizationVolts(int module, double value) {
+    for (int i = 0; i < 4; i++) {
+      m_modules[i].setDriveVoltage(0);
+      if (module == i){
+        m_modules[i].setSteerVoltage(value);
+      } else {
+        m_modules[i].setSteerVoltage(0);
+      }
+    }
+  }
+  
+
+  public void setChassisSpeeds(ChassisSpeeds chassisSpeeds) {
+    m_swerveModuleStates = m_kinematics.toSwerveModuleStates(chassisSpeeds);
+    SwerveDriveKinematics.desaturateWheelSpeeds(m_swerveModuleStates, DriveConstants.kMaxSpeed);
+    setModuleStates(m_swerveModuleStates);
+  }
+
+  public PIDController getXController() {
+      return m_xController;
+  }
+  public PIDController getYController() {
+      return m_yController;
+  }
+  public PIDController getRotationController() {
+    return m_rotationController;
+  }
+  public void setAllOptimize(Boolean optimizeSate){
+    for (int i = 0; i < 4; i++) {
+      m_modules[i].setOptimize(optimizeSate);
+    }
+  }
+  public void setupDrivetrainShuffleboard() {
+    // inputs
+    m_heading = m_drivetrainTab.add("Set Heading (-pi to pi)", 0).getEntry();
+    
+    // add PID controlers
+    m_drivetrainTab.add("xController", getXController());
+    m_drivetrainTab.add("yController", getYController());
+    m_drivetrainTab.add("rotationController", getRotationController());
+
+    m_drivetrainTab.addNumber("getAngle", () -> getAngleHeading());
+    m_drivetrainTab.addNumber("heading PID output", () -> m_headingPIDOutput);
+
+    m_drivetrainTab.addNumber("Gyro X", () -> getAngularRate(0));
+    m_drivetrainTab.addNumber("Gyro Y", () -> getAngularRate(1));
+    m_drivetrainTab.addNumber("Gyro Z", () -> getAngularRate(2));
+
+    // m_drivetrainTab.add("odometry", m_odometry);
+
+    m_drivetrainTab.add(getXController());
+    m_drivetrainTab.add(getYController());
+    m_drivetrainTab.add(getRotationController());
+  }
+  public void setupModulesShuffleboard(){
+    setUpModuleChooser();
+    setUpFeedforwardSavers();
+    
+    // inputs
+    m_swerveModulesTab.add("Module Chooser", m_moduleChooser);
+    m_driveVelocity = m_swerveModulesTab.add("Set Drive Velocity", 0).getEntry();
+    m_steerVelocity = m_swerveModulesTab.add("Set Steer Velocity", 0).getEntry();
+    m_steerAngle = m_swerveModulesTab.add("Set Steer Angle", 0).getEntry();
+    m_drivetrainvolts = m_swerveModulesTab.add("Set Volts", 0).getEntry();
+    m_driveStaticFeedforward = m_swerveModulesTab.add("Drive kS FF", m_driveStaticFeedForwardSaver[m_moduleChooser.getSelected().getModuleType().getID()]).getEntry();
```

Comment:
Here and below: indent  is needed if function call is spread across multiple lines.

Reply:
are you saying that the commands hould be spread across multiple lines?

Reply:
I'm saying that if you split a single statement (in this case function call) across multiple lines, identation should be added to lines after the first: